src/patient-registration/form-manager.ts

```diff
@@ -0,0 +1,150 @@
+  static createIdentifiers(
+    values: FormValues,
+    patientUuidMap: object,
+    identifierTypes: Array<PatientIdentifierType>,
+    abortController: AbortController,
+    location: string,
+  ) {
+    const identifiers: Array<Promise<PatientIdentifier>> = identifierTypes.map(type => {
+      const idValue = values[type.fieldName];
+      if (idValue) {
+        return Promise.resolve({
+          uuid: patientUuidMap[type.fieldName] ? patientUuidMap[type.fieldName].uuid : undefined,
+          identifier: idValue,
+          identifierType: type.uuid,
+          location: location,
+          preferred: type.isPrimary,
+        });
+      } else if (type.autoGenerationSource) {
+        return new Promise((resolve, _) => {
+          generateIdentifier(type.autoGenerationSource.uuid, abortController).then(response => {
+            resolve({
+              // is this undefined?
+              uuid: undefined,
+              identifier: response.data.identifier,
+              identifierType: type.uuid,
+              location: location,
+              preferred: type.isPrimary,
+            });
+          });
+        });
+      }
+    });
+
+    return Promise.all(identifiers);
+  }
+
+  static populateAddressValues(values: FormValues, initialAddressFieldValues: Object): Record<string, string> {
+    return Object.keys(initialAddressFieldValues).reduce(
+      (memo, fieldName) => ({ ...memo, [fieldName]: values[fieldName] }),
+      {},
+    );
+  }
+
+  static getDeletedNames(patientUuidMap: PatientUuidMapType) {
+    if (patientUuidMap?.additionalNameUuid) {
+      return [
+        {
+          nameUuid: patientUuidMap.additionalNameUuid,
+          personUuid: patientUuidMap.patientUuid,
+        },
+      ];
+    }
+    return [];
+  }
+
+  static getNames(values: FormValues, patientUuidMap: PatientUuidMapType) {
+    const names = [
+      {
+        uuid: patientUuidMap['preferredNameUuid'],
+        preferred: true,
+        givenName: values.givenName,
+        middleName: values.middleName,
+        familyName: values.familyName,
+      },
+    ];
+    if (values.addNameInLocalLanguage) {
+      names.push({
+        uuid: patientUuidMap['additionalNameUuid'],
+        preferred: false,
+        givenName: values.additionalGivenName,
+        middleName: values.additionalMiddleName,
+        familyName: values.additionalFamilyName,
+      });
+    }
+
+    return names;
+  }
+
+  static getAfterUrl(patientUuid: string, search: string, config: ConfigObject) {
+    return (
+      new URLSearchParams(search).get('afterUrl') ||
+      interpolateString(config.links.submitButton, { patientUuid: patientUuid })
+    );
+  }
+
+  static getDeathInfo(values: FormValues) {
+    const patientIsDead = {
+      dead: true,
+      deathDate: values.deathDate,
+      causeOfDeath: values.deathCause,
+    };
+
+    const patientIsNotDead = { dead: false };
+
+    return values.isDead ? patientIsDead : patientIsNotDead;
```

Comment:
Even though we only need 1 object we always create 2. Can we optimize?

Reply:
(Also `patientIsNotDead` could be module-globally stored - its a fixed object that never changes if I see this correctly.)